server/aesdsocket.c

```diff
@@ -102,224 +121,601 @@ int main(int argc, char **argv)
+
+void *socketThreadProcessing(void *ptr)
+{
+
+    int totalBuffer = BUFFER_STD_SIZE;
+    int currentSize = 0;
+    char *writebuffer = NULL;
+    char *readFromFile = NULL;
+    int bytesToBeWritten = 0;
+    int lseekValue = 0;
+    int bytesToBeRead = 0;
+    int recvClient = 0;
+    char bufferClientReceive[BUFFER_STD_SIZE];
+
+    /*Thread Parameters*/
+    struct threadSocketSt *threadParam = (struct threadSocketSt *)ptr;
+
+    writebuffer = (char *)malloc(sizeof(char) * BUFFER_STD_SIZE);
+    if (writebuffer == NULL)
+    {
+        perror("[-] Malloc for Write Buffer Failed\n");
+        state = THREAD_EXIT;
     }
 
-    close(fd);
-    close(serverSocket);
-    remove("/var/tmp/aesdsocketdata");
+    pthread_mutex_lock(&mutexSocket);
+    mallocCounter2++;
+    syslog(LOG_INFO, "Malloc Counter 2 : %d\n", mallocCounter2);
+    pthread_mutex_unlock(&mutexSocket);
+    // mallocCounter2++;
 
-    close(STDOUT_FILENO);
-    close(STDIN_FILENO);
-    close(STDERR_FILENO);
+    state = RECV_CLIENT;
 
-    syslog(LOG_INFO, "Signal Handler Completed\n");
+    /*Implement state Machine*/
+    while (1)
+    {
+        switch (state)
+        {
+        case RECV_CLIENT:
+            syslog(LOG_INFO, "STATE 7: RECV_CLIENT\n");
+            bool exitLoop = true;
+
+            while (exitLoop)
+            {
+                recvClient = recv(threadParam->clientSocket, bufferClientReceive, BUFFER_STD_SIZE, 0);
+
+                if (recvClient == 0 || (strchr(bufferClientReceive, '\n') != NULL))
+                {
+                    syslog(LOG_INFO, "Packet Completed\n");
+                    exitLoop = false;
+                }
+
+                if ((totalBuffer - currentSize) < recvClient)
+                {
+                    totalBuffer += recvClient;
+                    writebuffer = (char *)realloc(writebuffer, sizeof(char) * totalBuffer);
+                }
+
+                memcpy(writebuffer + currentSize, bufferClientReceive, recvClient);
+                currentSize += recvClient;
+            }
+
+            state = WRITE_DATA_TO_FILE;
+            break;
+
+        case WRITE_DATA_TO_FILE:
+            syslog(LOG_INFO, "STATE 8: WRITE_DATA_TO_FILE\n");
+
+            pthread_mutex_lock(&mutexSocket);
+            bytesToBeWritten = write(threadParam->fileDescriptor, writebuffer, currentSize);
+            lseekValue = lseek(threadParam->fileDescriptor, 0, SEEK_SET);
+            pthread_mutex_unlock(&mutexSocket);
+
+            state = READ_DATA_FROM_FILE;
+            break;
+
+        case READ_DATA_FROM_FILE:
+            syslog(LOG_INFO, "STATE 9: READ_DATA_FROM_FILE\n");
+            pthread_mutex_lock(&mutexSocket);
+            int lastCharOfFile = lseek(threadParam->fileDescriptor, 0, SEEK_END);
+            pthread_mutex_unlock(&mutexSocket);
+
+            //printf("Last Char of the File = %d*****************\n", lastCharOfFile);
+
+            readFromFile = (char *)malloc(sizeof(char) * lastCharOfFile);
```

Comment:
@dhirajbennadi what assumption are you making here about the total size of all packets sent?